lib/Request.js

```
@@ -310,3 +313,89 @@ Request.prototype.propertyError = function(property, errorCode, httpStatusCode,
+                return;
+            }
+
+            found = true;
+
+            // Aggregrate the data as it is streamed in
+            var chunks = [];
+
+            file.on('data', function(data) {
+                chunks.push(data)
+            });
+
+            file.on('end', function() {
+                resolve({
+                    buffer: Buffer.concat(chunks),
+                    fieldname: uploadedFieldName,
+                    encoding: encoding,
+                    mimetype: mimetype,
+                    filename: filename
+                });
+            });
+        });
+
+        stream.on('finish', function() {
+            if (!found) {
+                reject(new Error('Did not find uploaded file under field name ' + field));
+            }
+        });
+    }.bind(this));
+};
+
+Request.prototype.getAllUploads = function(field) {
+    return new Promise(function(resolve, reject) {
+        var stream = this.getStream();
+        var found = [];
+
+        stream.on('file', function(uploadedFieldName, file, filename, encoding, mimetype) {
+            if (field && field != uploadedFieldName) {
+                // Got an upload, but not for the field name we are looking for.
+                // Skip this upload, maybe the next one will be under the desired field name.
+                return;
+            }
+
+            // Aggregrate the data as it is streamed in
+            var chunks = [];
+
+            file.on('data', function(data) {
+                chunks.push(data);
+            });
+
+            file.on('end', function() {
+                found.push({
+                    buffer: Buffer.concat(chunks),
+                    fieldname: uploadedFieldName,
+                    encoding: encoding,
+                    mimetype: mimetype,
+                    filename: filename
+                });
+            });
+        });
+
+        stream.on('finish', function() {
+            resolve(found);
+        });
```

do we not want to throw an error when it's not found?